test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package test contains k8s integration tests
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edwarnicke/exechelper"
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/networkservicemesh/integration-k8s-kind/k8s"
+)
+
+const (
+	prefixesFilePath    = "/var/lib/networkservicemesh/config/excluded_prefixes.yaml"
+	collectorNamespace  = "excluded-prefixes-collector"
+	excludedPrefixesEnv = "EXCLUDE_PREFIXES_K8S_EXCLUDED_PREFIXES"
+)
+
+type ExcludedPrefixesSuite struct {
```

Comment:
Can we also verify that nsm prefixes config map is immutable for k8s admin?

Reply:
Added test for nsm-config external change catch